lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1239,7 +1239,7 @@ spawn_background(void)
 }
 #endif /* OFFLOAD_QUARANTINE */
 
-__attribute__((constructor))
+__attribute__((__constructor__(100)))
```

Comment:
Yeah I don't know what the right value to use is. In theory parts of the implementation could be wanting to free in early constructors. And no matter where you put it, the executable's .preinit_array runs before any constructors, with .preinit_array *only* existing for the executable, so there is no way for libc.so to ensure this runs before the executable runs .preinit_array, which I guess could free things. jemalloc's approach to its own initialisation is to check on entry if it needs to initialise itself with a `__predict_false(...)`, perhaps because of this problem, or perhaps because it needs to work as a drop-in library that can't assume it's part of the implementation. Mirroring that may be the best path forward.